hr_expense_petty_cash/models/hr_expense.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    payment_mode = fields.Selection(
+        selection_add=[('petty_cash', 'Petty Cash')],
+    )
+    petty_cash_id = fields.Many2one(
+        string='Petty cash holder',
+        comodel_name='petty.cash',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    def action_submit_expenses(self):
+        res = super().action_submit_expenses()
+        context = res.get('context', False)
+        context['default_expense_line_ids'] = self.ids
+        context['default_name'] = self[0].name if len(self) == 1 else ''
+        set_petty_cash_id = set()
+        for rec in self:
+            if rec.payment_mode == 'petty_cash':
+                set_petty_cash_id.add(rec.petty_cash_id.id)
+        if len(set_petty_cash_id) > 1:
```

Comment:
What if they use the same petty cash holder, but create multiple petty cash expense? I think, it should be ok, right?